File: Clase semana 07/02 - Clases grabadas/2024/1C/Clase07-Lunes/src/Lista1/Nodo.cpp
/*
 * Nodo.cpp
 *
 *  Created on: 22/04/2024
 *      Author: algo2
 */

#include "Nodo.h"
#include <iostream>

Nodo::Nodo(int dato) {
	this->dato = dato;
	this->siguiente = NULL;
}

Nodo::~Nodo() {}


int Nodo::getDato() {
	return this->dato;
}

void Nodo::setDato(int dato) {
	this->dato = dato;
}

const Nodo* Nodo::getSiguiente() {
	return this->siguiente;
}

void setSiguiente(Nodo* siguiente) {
	this->siguiente = siguiente;
}

File: Clase semana 07/02 - Clases grabadas/2024/2C/src/ar/uba/fi/cb100/semana07/miercoles/Nodo.java
package ar.uba.fi.cb100.semana07.miercoles;

public class Nodo<T> {
	//ATRIBUTOS DE CLASE --------------------------------------------------------------------------------------
	//ATRIBUTOS -----------------------------------------------------------------------------------------------

	private T dato;
	private Nodo<T> siguiente = null;

	//CONSTRUCTORES -------------------------------------------------------------------------------------------

	/**
	 * pre: -
	 * pos: el Nodo resulta inicializado con el dato dado
     *       y sin un Nodo siguiente.
	 */
	Nodo(T dato) {
		this.dato = dato;
		this.siguiente = null;
	}

	//METODOS DE CLASE ----------------------------------------------------------------------------------------
	//METODOS GENERALES ---------------------------------------------------------------------------------------
	//METODOS DE COMPORTAMIENTO -------------------------------------------------------------------------------
	//GETTERS SIMPLES -----------------------------------------------------------------------------------------

	/**
	 * pre:
	 * pos: devuelve el dato del nodo
	 */
	T getDato() {
		return this.dato;
	}

	/**
	 * pre:
	 * pos: cambia el dato del nodo
	 */
	Nodo<T> getSiguiente() {
		return this.siguiente;
	}

	//SETTERS SIMPLES -----------------------------------------------------------------------------------------	

	/**
	 * pre:
	 * pos: devuelve el siguiente nodo
	 */
	void setDato(T dato) {
		this.dato = dato;
	}
	
	/**
	 * pre:
	 * pos: cambia el nodo siguiente
	 */
	void setSiguiente(Nodo<T> siguiente) {
		this.siguiente = siguiente;
	}

}